### ISSUE TITLE
@template shouldn't be blocked by no-redundant-jsdoc

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3386]
### Bug Report

- __TSLint version__: 5.8
- __TypeScript version__: 2.6
- __Running TSLint via__:  VSCode

#### TypeScript code being linted

```typescript
/**
 * Props for a model view.
 *
 * @type TStore   Type of the backing model store.
 */
export interface IModelViewProps<TStore extends ModelStore> {
	/**
	 * Backing model store.
	 */
	store: TStore;
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-redundant-jsdoc": true
    }
}
```

#### Actual behavior

```shell
file: 'file:///d%3A/Code/ClientV2/src/Factories/ModelViewFactory.tsx'
severity: 'Error'
message: 'JSDoc tag '@type' is redundant in TypeScript code. (no-redundant-jsdoc)'
at: '5,5'
source: 'tslint'
```

#### Expected behavior

How else do you document the generic type?

**Comment by andy-ms** [comment]
I think you mean `@template`? `@type` provides the type to a variable.

**Comment by JoshuaKGoldberg** [comment]
Works for me!

**Comment by JoshuaKGoldberg** [comment]
Err. `@template` is also blocked.
